addons/sale_coupon/models/sale_order.py

```diff
@@ -187,6 +223,7 @@ def _get_reward_values_discount(self, program):
             currently_discounted_amount = 0
             for line in lines:
                 discount_line_amount = min(self._get_reward_values_discount_percentage_per_line(program, line), amount_total - currently_discounted_amount)
+                discount_line_amount = min(self._get_reward_values_discount_percentage_per_tax(program, line.tax_id), discount_line_amount)
```

Comment:
I am worried about the performance as this could mean potentially `O(n^2n)` if I'm not mistaken.

Though I would expect most orders to not have a lot of order lines and/or discounts applied to them so this might be an invalid concern.